static/app/utils/replays/replayReader.tsx

```diff
@@ -45,7 +46,21 @@ export default class ReplayReader {
       return null;
     }
 
-    return new ReplayReader({attachments, replayRecord, errors});
+    try {
+      return new ReplayReader({attachments, replayRecord, errors});
+    } catch (err) {
+      Sentry.captureException(err);
+
+      // If something happens then we don't really know if it's the attachments
+      // array or errors array to blame (it's probably attachments though).
+      // Either way we can use the replayRecord to show some metadata, and then
+      // put an error message below it.
+      return new ReplayReader({
+        attachments: [],
+        errors: [],
+        replayRecord,
+      });
+    }
```

Comment:
This is the important try/catch

in the failure case we still return a valid ReplayReader instance, albeit with less data than we expected.

Reply:
we've got that ui, checkout the screen shot in the desc.

what happens is we mark `loading=finished` and then there are not enough rrweb events, so we show that message. We are not showing a message with the contents of the caught error. The reason though would be malformed data that wasn't parsed correctly, which could be an SDK payload thing, like if the customer is messing with the sdk output and gives us weird stuff, the server doesn't care about the schema and would store it anyway (at least in the pre pii stripping days).